apps/condo/domains/billing/schema/SearchBillingReceiptsWithoutConsumerService.js

```diff
@@ -0,0 +1,158 @@
+                const { data } = args
+                const {
+                    residents: residentIds,
+                } = data
+
+                const uniqueResidentIds = new Set(residentIds.map(resident => resident.id))
+                if (residentIds.length !== uniqueResidentIds.size) {
+                    throw new GQLError(errors.DUPLICATED_RESIDENTS, context)
+                }
+
+                // firstly try to retrieve all residents passed to the service
+                const residents = await Resident.getAll(context, {
+                    id_in: uniqueResidentIds,
+                    user: { id: context.authedItem.id },
+                    deletedAt: null,
+                })
+
+                if (residents.length !== residentIds.length) {
+                    throw new GQLError(errors.MISSING_RESIDENTS, context)
+                }
+
+                // retrieve related ServiceConsumers
+                const consumers = await ServiceConsumer.getAll(context, {
+                    resident: { id_in: uniqueResidentIds },
+                    deletedAt: null,
+                })
+
+                // filter residents that have service consumer
+                const residentsWithoutConsumer = residents.filter(resident => {
+                    const exists = consumers.find(consumer => consumer.resident.id === resident.id)
+                    return !exists
+                })
+
+                // we have to proceed search resident by resident
+                const residentReceipts = []
+                for (let i = 0 ; i < residentsWithoutConsumer.length; i++) {
+                    const {
+                        unitName,
+                        unitType,
+                        property: {
+                            id: propertyId,
+                        },
+                    } = residentsWithoutConsumer[i]
+
+                    const property = await Property.getOne(context, { id: propertyId, deletedAt: null }, {
+                        doesNotExistError: errors.MISSING_RESIDENT_PROPERTY,
+                    })
+
+                    // assemble conditions
+                    const addressCondition = {
+                        OR: [
+                            { property: { address: property.address } },
+                            { property: { addressKey: property.addressKey } },
+                            { property: { normalizedAddress: property.address } },
+                        ],
+                    }
+                    const accountCondition = {
+                        account: {
+                            unitName,
+                            unitType,
+                        },
+                    }
+                    const integrationCondition = {
+                        context: {
+                            status: CONTEXT_FINISHED_STATUS,
+                        },
+                    }
+
+                    const receipts = await BillingReceipt.getAll(context, {
```

Comment:
if there are lots of receipts, better to use something like loadListByChunks or add pagination

Reply:
Now it becomes BillingAccount entity searching only for newly registered of condo platform. It is not a big result we are expecting as a result.

Reply:
But probably you are right, better to support all cases

Reply:
fixed